# Mini Project Modul 3: Menganalisis Hasil A/B Test untuk Desain Website Baru

**Skenario Bisnis**

Perusahaan Anda telah meluncurkan A/B test untuk desain homepage baru. Tujuannya adalah untuk melihat apakah desain baru (treatment) membuat pengguna menghabiskan lebih banyak waktu di situs dibandingkan dengan desain lama (control).
- Grup A (Control): Melihat desain homepage lama.
- Grup B (Treatment): Melihat desain homepage baru.

Pertanyaan: Apakah desain baru secara signifikan meningkatkan rata-rata durasi sesi pengguna?

## Langkah 1: Menyiapkan Data & Hipotesis

### a. Buat Dataset (Menggunakan Python)

In [1]:
import pandas as pd
import numpy as np

# Membuat data tiruan
np.random.seed(42) # Agar hasilnya selalu sama
data = {
    'user_id': range(1, 2001),
    'group': ['control'] * 1000 + ['treatment'] * 1000,
    'session_duration_minutes': np.concatenate([
        np.random.normal(loc=10, scale=3, size=1000), # Data untuk grup control
        np.random.normal(loc=10.8, scale=3.2, size=1000) # Data untuk grup treatment
    ])
}
df = pd.DataFrame(data)

# Menyimpan ke file CSV
df.to_csv('ab_test_data.csv', index=False)

print("File 'ab_test_data.csv' berhasil dibuat!")

File 'ab_test_data.csv' berhasil dibuat!


### b. Definisikan Hipotesis
- Hipotesis Nol (H₀): Tidak ada perbedaan. Rata-rata durasi sesi grup treatment sama dengan grup control.
- Hipotesis Alternatif (H₁): Ada perbedaan. Rata-rata durasi sesi grup treatment lebih tinggi daripada grup control.
- Tingkat Signifikansi (α): Kita akan menggunakan standar 0.05.

## Langkah 2: Analisis Data

### Muat dan Lihat Data

In [2]:
import pandas as pd

# Muat dataset
df = pd.read_csv('ab_test_data.csv')

# Pisahkan data menjadi dua grup
control_group = df[df['group'] == 'control']['session_duration_minutes']
treatment_group = df[df['group'] == 'treatment']['session_duration_minutes']

### b. Hitung Statistik Deskriptif

In [3]:
# Menghitung rata-rata dan standar deviasi
print(f"Rata-rata Control: {control_group.mean():.2f}")
print(f"Rata-rata Treatment: {treatment_group.mean():.2f}")
print(f"Std Deviasi Control: {control_group.std():.2f}")
print(f"Std Deviasi Treatment: {treatment_group.std():.2f}")

Rata-rata Control: 10.06
Rata-rata Treatment: 11.03
Std Deviasi Control: 2.94
Std Deviasi Treatment: 3.19


## Langkah 3: Lakukan Uji Statistik (T-test)

In [4]:
from scipy import stats

# Melakukan independent t-test
t_stat, p_value = stats.ttest_ind(treatment_group, control_group, alternative='greater')

print(f"T-statistic: {t_stat:.4f}")
print(f"P-value: {p_value:.4f}")

# Membuat keputusan
alpha = 0.05
if p_value < alpha:
    print("\nHasil signifikan! Kita menolak Hipotesis Nol.")
else:
    print("\nHasil tidak signifikan. Kita gagal menolak Hipotesis Nol.")

T-statistic: 7.0615
P-value: 0.0000

Hasil signifikan! Kita menolak Hipotesis Nol.


## Langkah 4: Interpretasi dan Kesimpulan

### a. Interpretasi Hasil Statistik
Hasil yang didapat adalah:
- T-statistic: 7.0615
- P-value: 0.0000
P-value yang didapat adalah (0.000) sangat kecil, jauh di bawah tingkat signifikansi (alpha = 0.05)

### b. Kesimpulan
Berdasarkan analisis statistik, didapat bukti yang sangat kuat untuk **menolak Hipotesis Nol.**

*Rekomendasi untuk Manajemen:*

> Hasil dari A/B test menunjukkan bahwa desain homepage yang baru secara signifikan meningkatkan rata-rata durasi sesi pengguna dibandingkan dengan desain lama (p < 0.05). Rata-rata durasi sesi untuk desain baru adalah sekitar 11.03 menit, lebih tinggi dari desain lama yang rata-ratanya 10.06 menit. Direkomendasikan untuk meluncurkan desain homepage baru ini kepada semua pengguna untuk meningkatkan engagement.